lib/logstash/outputs/elasticsearch/http_client/pool.rb

```diff
@@ -237,7 +270,7 @@ def perform_request_to_url(url, method, path, params={}, body=nil)
     end
 
     def normalize_url(uri)
-      raise ArgumentError, "Only URI objects may be passed in!" unless uri.is_a?(URI)
+      raise ArgumentError, "Only SafeURI objects may be passed in!" unless uri.is_a?(URI) || uri.is_a?(LogStash::Util::SafeURI)
```

Comment:
This allows URI and SafeURI, but the exception indicates only SafeURI is accepted. Confusing?